src/core_plugins/timelion/public/vis/timelion_request_handler.js

```diff
@@ -12,28 +12,20 @@ const TimelionRequestHandlerProvider = function (Private, Notifier, $http, $root
 
   return {
     name: 'timelion',
-    handler: function (vis /*, appState, uiState, queryFilter*/) {
+    handler: function (vis, { timeRange }) {
```

Comment:
Make timelion request handler only use the explicitly passed `timeRange` instead of accessing `timefilter.time` directly.